pkg/proxy/winkernel/hnsV2.go

```diff
@@ -232,16 +232,37 @@ func (hns hnsV2) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFl
 		lbFlags |= hcn.LoadBalancerFlagsDSR
 	}
 
-	lb, err := hcn.AddLoadBalancer(
-		hnsEndpoints,
-		lbFlags,
-		lbPortMappingFlags,
-		sourceVip,
-		vips,
-		protocol,
-		internalPort,
-		externalPort,
-	)
+	lbDistributionType := hcn.LoadBalancerDistributionNone
+
+	if flags.sessionAffinity {
+		lbDistributionType = hcn.LoadBalancerDistributionSourceIP
+	}
+
+	loadBalancer := &hcn.HostComputeLoadBalancer{
+		SourceVIP: sourceVip,
+		PortMappings: []hcn.LoadBalancerPortMapping{
+			{
+				Protocol:         uint32(protocol),
+				InternalPort:     internalPort,
+				ExternalPort:     externalPort,
+				DistributionType: lbDistributionType,
+				Flags:            lbPortMappingFlags,
+			},
+		},
+		FrontendVIPs: vips,
+		SchemaVersion: hcn.SchemaVersion{
+			Major: 2,
```

Comment:
Formatting

Reply:
Are you referring to the lack of space before the curly bracket? If so, I think this is correct. When I add a space, hack/update-gofmt.sh removes it.

Reply:
Oh I just missed that Major, Minor fields were part of the struct, hence the indentation. Ignore this comment